From c22bd5873f789b7646ce2a25854145cc10a2306a Mon Sep 17 00:00:00 2001
From: Rob Clark <rclark@redhat.com>
Date: Wed, 17 Dec 2014 21:38:52 -0500
Subject: [drm] radeon: fix semaphore value init

Message-id: <1418852380-13061-42-git-send-email-rclark@redhat.com>
Patchwork-id: 102178
O-Subject: [RHEL7 drm 41/89] drm/radeon: fix semaphore value init
Bugzilla: 1173317
RH-Acked-by: Jerome Glisse <jglisse@redhat.com>
RH-Acked-by: Dave Airlie <airlied@redhat.com>

Bugzilla: https://bugzilla.redhat.com/show_bug.cgi?id=1173317
Upstream: since v3.16.4

commit 007d978abcdbb25bd138aa0f4c075274628a5e5d

Author:     Christian Koenig <christian.koenig@amd.com>
AuthorDate: Sun Sep 7 12:06:52 2014 +0200
Commit:     Rob Clark <rclark@redhat.com>
CommitDate: Wed Dec 10 17:19:52 2014 -0500

    drm/radeon: fix semaphore value init

    commit f229407da79315c18a2f25f485e1a1b9fdda1e92 upstream.

    Semaphore values have 64 bits, not 32. This fixes a very subtle bug
    that disables synchronization when the upper 32bits wasn't zero.

    Signed-off-by: Christian Koenig <christian.koenig@amd.com>
    Reviewed-By: Grigori Goronzy <greg@chown.ath.cx>
    Signed-off-by: Alex Deucher <alexander.deucher@amd.com>
    Signed-off-by: Greg Kroah-Hartman <gregkh@linuxfoundation.org>

Signed-off-by: Rob Clark <rclark@redhat.com>
Signed-off-by: Jarod Wilson <jarod@redhat.com>

diff --git a/drivers/gpu/drm/radeon/radeon_semaphore.c b/drivers/gpu/drm/radeon/radeon_semaphore.c
index dbd6bcd..e6101c1 100644
--- a/drivers/gpu/drm/radeon/radeon_semaphore.c
+++ b/drivers/gpu/drm/radeon/radeon_semaphore.c
@@ -34,7 +34,7 @@
 int radeon_semaphore_create(struct radeon_device *rdev,
        struct radeon_semaphore **semaphore)
 {
- uint32_t *cpu_addr;
+ uint64_t *cpu_addr;
  int i, r;
 
  *semaphore = kmalloc(sizeof(struct radeon_semaphore), GFP_KERNEL);
-- 
1.7.1